pkg/manager/manager.go

```diff
@@ -46,16 +49,45 @@ type Status struct {
 	WorkerStatus map[string]worker.Status
 }
 
+// fromCheckpoint laods last invoke time from json
+func fromCheckpoint(checkpointFile string) (map[string]time.Time, error) {
```

Comment:
Extract checkpoint struct (`map[string]time.Time`) into a struct for further extensibility:

```
type WorkerCheckPoint struct {
    LastInvokeTime time.Time
}

type CheckPoint struct {
    workerInfo map[string]WorkerCheckPoint
}
```